src/lib/device/round_robin_chunk_selector.h

```diff
@@ -43,14 +35,12 @@ class RoundRobinChunkSelector : public ChunkSelector {
     RoundRobinChunkSelector& operator=(RoundRobinChunkSelector&&) noexcept = delete;
     ~RoundRobinChunkSelector() = default;
 
-    void add_chunk(cshared< Chunk >& chunk) override;
-
+    void add_chunk(VChunk chunk) override;
     Chunk* select(blk_count_t nblks, const blk_alloc_hints& hints) override;
-
-    void foreach_chunks(std::function< void(cshared< Chunk >&) >&& cb) override;
+    void foreach_chunks(std::function< void(VChunk&) >&& cb) override;
 
 private:
-    std::vector< shared< Chunk > > m_chunks;
+    std::vector< VChunk > m_chunks;
```

Comment:
If the decision is to make VChunk a base/abstract class, it needs to be vector of shared_ptr of VChunk (you probably are already aware of this).

Reply:
A `shared_ptr<VChunk>` does not make any sense; it's simply a shared_ptr itself; it's move/copy friendly so it should never be wrapped in a pointer and passed by reference or move semantics.

Reply:
@szmyd you misunderstood my comment, which is based on VChunk itself doesn't contain a cshared <Chunk>, but have Chunk derived from VChunk. VChunk will become an abstract class, and m_chunks will be pointing to the base class and use dynamic_cast in run_time to get its real type. 

I am not suggesting we should do this way, that why the comment started with an "if"